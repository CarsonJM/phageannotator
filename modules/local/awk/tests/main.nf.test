nextflow_process {

    name "Test Process: AWK"
    script "modules/local/awk/main.nf"
    process "AWK"
    tag "modules_local"
    tag "awk"


    test("Test AWK") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id: 'all_samples' ], [ file(params.test_data['modules_local']['virus_summary1_tsv'], checkIfExists: true), file(params.test_data['modules_local']['virus_summary2_tsv'], checkIfExists: true) ] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

        cleanup {
                println "Clean up..."
            path("$workDir").deleteDir()
        }
    }
}
