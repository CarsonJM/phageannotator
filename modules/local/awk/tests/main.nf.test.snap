{
    "Test AWK": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.tsv:md5,dc9989bca53c3886c48b849977890be0"
                    ]
                ],
                "1": [
                    "versions.yml:md5,de757fec2101a5dccf996912d9c791c7"
                ],
                "file_out": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.tsv:md5,dc9989bca53c3886c48b849977890be0"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de757fec2101a5dccf996912d9c791c7"
                ]
            }
        ],
        "timestamp": "2023-09-25T20:33:32+0000"
    }
}